libraries/standard/http/test/test-HTTPClient_InitializeRequestHeaders.c

```diff
@@ -0,0 +1,220 @@
+#define HTTP_TEST_HOST_VALUE            "amazon.com"
+#define HTTP_TEST_HOST_VALUE_LEN        ( sizeof( HTTP_TEST_HOST_VALUE ) - 1 )
+#define HTTP_TEST_REQUEST_LINE     \
+    ( HTTP_TEST_REQUEST_METHOD " " \
+      HTTP_TEST_REQUEST_PATH " "   \
+      HTTP_PROTOCOL_VERSION "\r\n" )
+#define HTTP_TEST_REQUEST_LINE_LEN      ( sizeof( HTTP_TEST_REQUEST_LINE ) - 1 )
+
+/* Used for format parameter in snprintf(...). */
+#define HTTP_TEST_HEADER_FORMAT \
+    "%s %s %s\r\n"              \
+    "%s: %s\r\n"                \
+    "%s: %s\r\n"                \
+    "%s: %s\r\n\r\n"
+
+#define HTTP_REQUEST_HEADERS_INITIALIZER    { 0 }
+#define HTTP_REQUEST_INFO_INITIALIZER       { 0 }
+
+/* Length of the following template HTTP header.
+ *   <HTTP_TEST_REQUEST_METHOD> <HTTP_TEST_REQUEST_PATH> <HTTP_PROTOCOL_VERSION> \r\n
+ *   <HTTP_USER_AGENT_FIELD>: <HTTP_USER_AGENT_FIELD_LEN> \r\n
+ *   <HTTP_HOST_FIELD>: <HTTP_TEST_HOST_VALUE> \r\n
+ *   <HTTP_CONNECTION_FIELD>: \r\n
+ *   \r\n
+ * This is used to initialize the expectedHeader string. Note the missing
+ * <HTTP_TEST_CONNECTION_VALUE>. This is added later on depending on the
+ * value of HTTP_REQUEST_KEEP_ALIVE_FLAG in reqInfo->flags. */
+#define HTTP_TEST_PREFIX_HEADER_LEN                                 \
+    ( HTTP_TEST_REQUEST_METHOD_LEN + SPACE_CHARACTER_LEN +          \
+      HTTP_TEST_REQUEST_PATH_LEN + SPACE_CHARACTER_LEN +            \
+      HTTP_PROTOCOL_VERSION_LEN + HTTP_HEADER_LINE_SEPARATOR_LEN +  \
+      HTTP_USER_AGENT_FIELD_LEN + HTTP_HEADER_FIELD_SEPARATOR_LEN + \
+      HTTP_USER_AGENT_VALUE_LEN + HTTP_HEADER_LINE_SEPARATOR_LEN +  \
+      HTTP_HOST_FIELD_LEN + HTTP_HEADER_FIELD_SEPARATOR_LEN +       \
+      HTTP_TEST_HOST_VALUE_LEN + HTTP_HEADER_LINE_SEPARATOR_LEN +   \
+      HTTP_CONNECTION_FIELD_LEN + HTTP_HEADER_FIELD_SEPARATOR_LEN + \
+      HTTP_HEADER_LINE_SEPARATOR_LEN +                              \
+      HTTP_HEADER_LINE_SEPARATOR_LEN )
+
+/* Add HTTP_CONNECTION_KEEP_ALIVE_VALUE_LEN to account for longest possible
+ * length of template header. */
+#define HTTP_TEST_MAX_HEADER_LEN \
+    ( HTTP_TEST_PREFIX_HEADER_LEN + HTTP_CONNECTION_KEEP_ALIVE_VALUE_LEN )
+
+/* Add 1 because snprintf(...) writes a null byte at the end. */
+#define HTTP_TEST_BUFFER_SIZE    ( HTTP_TEST_MAX_HEADER_LEN + 1 )
+
+
+int main()
+{
+    HTTPRequestHeaders_t reqHeaders = HTTP_REQUEST_HEADERS_INITIALIZER;
+    HTTPRequestHeaders_t reqHeadersDflt = HTTP_REQUEST_HEADERS_INITIALIZER;
+    HTTPRequestInfo_t reqInfo = HTTP_REQUEST_INFO_INITIALIZER;
+    HTTPRequestInfo_t reqInfoDflt = HTTP_REQUEST_INFO_INITIALIZER;
+    HTTPStatus_t test_err = HTTP_NOT_SUPPORTED;
+    uint8_t buffer[ HTTP_TEST_BUFFER_SIZE ] = { 0 };
+    char expectedHeader[ HTTP_TEST_BUFFER_SIZE ] = { 0 };
+    size_t expectedHeaderLen = HTTP_TEST_MAX_HEADER_LEN;
+
+/* Write template reqInfo to pass as parameter to
+ * HTTPClient_InitializeRequestHeaders() method. */
+#define fillReqInfoTemplate()                             \
```

Comment:
These complex inline macros, defined within a local function scope, are atypical and tough to follow. What's the rationale?

Reply:
When we move to Unity this test setup will move to a TEST_SETUP(){} macro area. This is to setup the shared variable for each of the tests separated by `/* -----------------------------------------------------------------------*/`.

This method comes from Dan's testing framework: https://gist.github.com/dan4thewin/6f708bf635f6cb647d9f7bc7f55e4706#the-complete-unit-test

Reply:
This is one of the reasons we decided against this method moving forward, CMock does not require you to do these and as such this will be resolved as part of the conversion. For now this is fine as it is in the tets only and not in the library code...